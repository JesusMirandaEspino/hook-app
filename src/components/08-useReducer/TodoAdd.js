import React from 'react'
import { UseForm } from '../../Hooks/UseForm';

export const TodoAdd = ( { handleAddTodo } ) => {

    const [  { description } , handleInpuntChange, reset ]  = UseForm( {
        description: ''
    } );


    const handleSubmit = (e) => {
        e.preventDefault();

        if( description.trim() <= 1 ){
            return;
        }

        const newTodo = {
            id: new Date().getTime(),
            desc: description,
            done: false
        };

        handleAddTodo( newTodo );
        reset();

    }

    return (
        <>
            <form  onSubmit={ handleSubmit } >

                <label for="description" > Ingresa un elemento </label> 
                <input type="text"  name="description" onChange={ handleInpuntChange }  value={ description }  />

                <button type="submit" className="btn btn-primary mt-1" >Agregar</button>

            </form>            
        </>
    )
}
