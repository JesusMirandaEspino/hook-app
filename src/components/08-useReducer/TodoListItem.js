import React from 'react'

export const TodoListItem = ( { todo, index, HandleDelete, handleToggle }  ) => {
    return (
            <li key={ todo.id }  className="list-group-item" >
                <p className={  ` ${ todo.done && 'complete' } ` } onClick={ () => handleToggle(todo.id) }  >  { index + 1 } .- {  todo.desc } </p>
                
                <button className="btn btn-danger" onClick={ () => HandleDelete(todo.id) }    >Borrar</button>

            </li>
    )
}
