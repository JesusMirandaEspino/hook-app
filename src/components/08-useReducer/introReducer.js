const initialState = [{
    id: 1,
    todo: 'Curso Angular',
    done: false
}];

const todoReducer = ( state = initialState, action ) => {

    if( action.type == 'agregarTodoAction' ){}

    return state;
};


let todos = todoReducer();

const newTodo = {
    id: 2,
    todo: 'Curso React',
    done: false
};

const agregarTodoAction = {
    type: 'agregar',
    payload: newTodo
};


todos = todoReducer(todos, agregarTodoAction);