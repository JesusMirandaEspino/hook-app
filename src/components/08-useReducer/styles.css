body{
    padding:50px;
}


.complete{
    text-decoration: line-through;
}

p{
    cursor: pointer;
}