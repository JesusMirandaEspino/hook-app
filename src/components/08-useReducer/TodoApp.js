import React, { useReducer } from 'react';
import { TodoReducer } from './TodoReducer'
import './styles.css';

export const TodoApp = () => {

    const initialState = [{

        id: new Date().getTime(),
        desc: 'Aprender React',
        done: false
    }];


    const [  todo ] = useReducer( TodoReducer, initialState  );


    

    return (
        <div>
            <h1>Todo App</h1>
            <hr/>

            <ul>

                <li>Hola</li>
                <li>Mundo</li>
                <li>Como</li>
                <li>Estan</li>

            </ul>

        </div>
    )
}
