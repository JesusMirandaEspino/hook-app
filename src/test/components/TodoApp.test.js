import React from 'react'
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

import { TodoApp } from '../../components/08-useReducer/TodoApp';

    const wrappers = shallow(  <TodoApp /> );

describe( 'Pruebas en  <TodoApp />', () => {


    test( 'Debe de mostrarse correctamente', () => {


        expect( wrappers ).toMatchSnapshot();

    });


} );