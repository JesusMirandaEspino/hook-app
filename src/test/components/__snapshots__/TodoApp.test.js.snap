// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en  <TodoApp /> Debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Todo App  ( 
    0
      ) 
  </h1>
  <hr />
  <TodoAdd
    handleAddTodo={[Function]}
  />
  <hr />
  <TodoList
    HandleDelete={[Function]}
    handleToggle={[Function]}
    todos={Array []}
  />
</div>
`;
